osgp/platform/osgp-throttling-service/src/main/java/org/opensmartgridplatform/throttling/cleanup/PermitCleanUpJob.java

```diff
@@ -35,35 +39,66 @@
   @Value("#{T(java.time.Duration).parse('${cleanup.permits.time-to-live:PT1H}')}")
   private Duration timeToLive;
 
+  @Value("${cleanup.permits.batch.size:50}")
+  private int batchSize;
+
   @Override
   public void execute(final JobExecutionContext jobExecutionContext) {
-    final List<Permit> expiredPermits =
-        this.permitRepository.findByCreatedAtBefore(Instant.now().minus(this.timeToLive));
-    LOGGER.debug("Found {} permits to be cleaned", expiredPermits.size());
-    expiredPermits.forEach(
-        permit -> {
-          LOGGER.warn(
-              "Cleaning up permit (bts: {}/cell: {}), created on {} for client {}",
-              permit.getBaseTransceiverStationId(),
-              permit.getCellId(),
-              permit.getCreatedAt(),
-              permit.getClientId());
-          if (!this.segmentedNetworkThrottler.releasePermit(
-              permit.getThrottlingConfigId(),
-              permit.getClientId(),
-              permit.getBaseTransceiverStationId(),
-              permit.getCellId(),
-              permit.getRequestId())) {
-            /*
-             * The throttler returns true if the permit has been deleted from the database. If the
-             * throttler ever returns false, it has not deleted the permit, possibly because for
-             * some reason it was not present in memory. In any such case, make sure the permit that
-             * exceeded its time to live is deleted, so it wont be loaded if the throttling service
-             * is restarted.
-             */
-            LOGGER.warn("Releasing permit with SegmentedNetworkThrottler failed");
-            this.permitRepository.delete(permit);
-          }
-        });
+    if (this.batchSize < 1) {
+      LOGGER.warn(
+          "Not cleaning any permits, as batch size is configured at a value less than one: cleanup.permits.batch.size={}",
+          this.batchSize);
+      return;
+    }
+    final Instant createdAtBefore = Instant.now().minus(this.timeToLive);
+    final PageRequest pageRequest = PageRequest.of(0, this.batchSize, Sort.by("id").ascending());
```

Comment:
https://www.baeldung.com/jpa-limit-query-results#1first-ortop

Reply:
Sorting doesn't seem to add anything here, might even impact query time. Just get the first _x_ permits.